src/utils/searchForm.util.js

```diff
@@ -0,0 +1,72 @@
+const identifierUrl = 'http://hl7.org/fhir/sid/us-npi|';
+
+function getParamsFromReq(req) {
+  const queryStart = req.originalUrl.indexOf('?');
+  const searchParams =
+    queryStart > -1 ? new URLSearchParams(req.originalUrl.substring(queryStart + 1)) : '';
+  return searchParams ? Object.fromEntries(searchParams.entries()) : {};
+}
+
+const getFormData = (req, resourceName) => {
+  const params = getParamsFromReq(req);
+  const identifierParts = params.identifier ? params.identifier.split('|') : [];
+
+  const givenNameField = {
+    label: 'Given (Name)',
+    name: 'given',
+    value: params.given ? params.given : '',
+  };
+  const familyNameField = {
+    label: 'Family (Name)',
+    name: 'family',
+    value: params.family ? params.family : '',
+  };
+
+  let formData = [];
+
+  switch (resourceName) {
+    case 'Patient':
+      formData.push(givenNameField);
```

Comment:
Maybe break this into separate methods to reduce size of function and encapsulate logic if needed.

Reply:
Now in multiple methods